activation/activation.go

```diff
@@ -452,17 +467,22 @@ func (b *Builder) buildNIPostChallenge(ctx context.Context) error {
 }
 
 // UpdatePoETServer updates poet client. Context is used to verify that the target is responsive.
-func (b *Builder) UpdatePoETServer(ctx context.Context, target string) error {
-	b.log.With().Debug("request to update poet service", log.String("target", target))
-	client := b.poetClientInitializer(target)
-	// TODO(dshulyak) not enough information to verify that PoetServiceID matches with an expected one.
-	// Maybe it should be provided during update.
-	sid, err := client.PoetServiceID(ctx)
-	if err != nil {
-		return fmt.Errorf("%w: %v", ErrPoetServiceUnstable, err)
+func (b *Builder) UpdatePoETServers(ctx context.Context, endpoints []string) error {
+	b.log.With().Debug("request to update poet services", log.Strings("endpoints", endpoints))
+
+	clients := make([]PoetProvingServiceClient, len(endpoints))
+	for _, endpoint := range endpoints {
+		client := b.poetClientInitializer(endpoint)
+		// TODO(dshulyak) not enough information to verify that PoetServiceID matches with an expected one.
+		// Maybe it should be provided during update.
+		sid, err := client.PoetServiceID(ctx)
+		if err != nil {
+			return fmt.Errorf("%w: failed to query Poet '%s' for ID (%v)", ErrPoetServiceUnstable, endpoint, err)
```

Comment:
It's weird to see two errors wrapped simultaneously: `ErrPoetServiceUnstable` and `err`.

The only way that I see around that would be to make `ErrPoetserviceUnstable` a proper type:

```golang
type ErrPoetServiceUnstable struct {
     Err error
}

func (e *ErrPoetServiceUnstable) Error() string { return fmt.Sprintf("builder: poet service is unstable: %v", e.Err) }

func (e *ErrPoetServiceUnstable) Unwrap() error { return e.Err }
```

and then use it here like this:

```suggestion
			return &ErrPotServiceUnstable{Err: fmt.Errorf("failed to query Poet '%s': %w", endpoint, err) }
```

Reply:
They are not both wrapped. `ErrPoetServiceUnstable` is wrapped and `err` provides more details. I agree it became awkward. I will think about this. Thx